Data/Primitive/ByteArray.hs

```diff
@@ -126,22 +129,49 @@ newAlignedPinnedByteArray (I# n#) (I# k#)
 -- | Yield a pointer to the array's data. This operation is only safe on
 -- /pinned/ byte arrays allocated by 'newPinnedByteArray' or
 -- 'newAlignedPinnedByteArray'.
+--
+-- Prefer 'withByteArrayContents', which ensures that the array is not
+-- garbage collected while the pointer is being used.
 byteArrayContents :: ByteArray -> Ptr Word8
 {-# INLINE byteArrayContents #-}
 byteArrayContents (ByteArray arr#) = Ptr (byteArrayContents# arr#)
 
+-- | A composition of 'byteArrayContents' and 'keepAliveUnlifted'.
+-- The callback function must not return the pointer.
```

Comment:
Does this also require the array to be pinned (like `byteArrayContents` does)?

Reply:
Yes, the byte array must be pinned.

Reply:
I've documented the requirement that the byte array be pinned.